File: app/src/main/java/com/bq/zowi/models/commands/FallCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class FallCommand extends MovementCommand {
    private static final Command.Action[] FALL_ACTION = {Command.Action.FALL};
    private static final Command.Direction[] FALL_DIRECTIONS = new Command.Direction[0];
    private static final long FALL_DURATION = 2000;

    public FallCommand(Command.Action action) {
        setAction(action);
        setDuration(FALL_DURATION);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return FALL_ACTION;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction[] getAllowedDirections() {
        return FALL_DIRECTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new FallCommand(getAction());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/FallCommand.kt
package com.bq.zowi.models.commands

class FallCommand(action: Command.Action) : MovementCommand() {

    init {
        setAction(action)
        setDuration(FALL_DURATION)
    }

    override fun getAllowedActions(): Array<Command.Action> = FALL_ACTION
    override fun getAllowedDirections(): Array<Command.Direction> = emptyArray()
    override fun copy(): Command = FallCommand(getAction())

    companion object {
        private val FALL_ACTION = arrayOf(Command.Action.FALL)
        private const val FALL_DURATION = 2000L
    }
}
